gensim/models/word2vec.py

```diff
@@ -472,39 +473,106 @@ def create_binary_tree(self):
 
             logger.info("built huffman tree with maximum node depth %i", max_depth)
 
-    def build_vocab(self, sentences, keep_raw_vocab=False):
+    def build_vocab(self, sentences, keep_raw_vocab=False, scan_size=10000):
         """
         Build vocabulary from a sequence of sentences (can be a once-only generator stream).
         Each sentence must be a list of unicode strings.
 
         """
-        self.scan_vocab(sentences)  # initial survey
+        if isinstance(sentences, tuple):
```

Comment:
With the new logic in place, if sentences is a generator (which it shouldn't) it's the best way I found to prevent it, making sure the code is passing the unit test for the generator.

If someone has a better way to deal with that, I'll be more than happy to apply it!